File: frontend/gyssys_shop/lib/models/articulo.dart
class Articulo {
  final int? id;
  final String codigo;
  final String nombre;
  final int cantidad;
  final double precio;
  final String? descripcion;
  final String categoria;
  final String? imagenUrl;

  Articulo({
    this.id,
    required this.codigo,
    required this.nombre,
    this.cantidad = 0,
    this.precio = 0.0,
    this.descripcion,
    this.categoria = 'otro/a',
    this.imagenUrl,
  });

  factory Articulo.fromJson(Map<String, dynamic> json) {
    return Articulo(
      id: json['id'],
      codigo: json['codigo'],
      nombre: json['nombre'],
      cantidad: json['cantidad'] ?? 0,
      precio: (json['precio'] ?? 0.0).toDouble(),
      descripcion: json['descripcion'],
      categoria: json['categoria'] ?? 'otro/a',
      imagenUrl: json['imagen_url'] ?? 'assets/images/no-image.jpeg',
    );
  }

  Map<String, dynamic> toJson() => {
        'codigo': codigo,
        'nombre': nombre,
        'cantidad': cantidad,
        'precio': precio,
        'descripcion': descripcion,
        'categoria': categoria,
        'imagen_url': imagenUrl,
      };
}

File: backend/models/articulo.js
module.exports = (sequelize, DataTypes) => {
  const Articulo = sequelize.define('Articulo', {
    codigo: {
      type: DataTypes.STRING(255),
      allowNull: false,
      unique: true
    },
    nombre: {
      type: DataTypes.STRING(255),
      allowNull: false
    },
    cantidad: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0
    },
    precio: {
      type: DataTypes.FLOAT,
      allowNull: false,
      defaultValue: 0
    },
    descripcion: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    categoria: {
      type: DataTypes.ENUM('ropa', 'accesorios', 'cosmeticos', 'calzado', 'otro/a'),
      allowNull: false,
      defaultValue: 'otro/a'
    },
    imagen_url: {
      type: DataTypes.STRING,
      allowNull: true,
      defaultValue: 'assets/images/no-image.jpeg'
    },
    imagen_public_id: {
      type: DataTypes.STRING,
      allowNull: true
    }
  }, {
    tableName: 'articulos',
    timestamps: true,
    paranoid: true
  });

  return Articulo;
};